.eslintrc.json

```diff
@@ -1,3 +1,29 @@
 {
-  "extends": "next/core-web-vitals"
+    "env": {
+        "browser": true,
+        "es2021": true
+    },
+    "extends": [
+        "plugin:react/recommended",
+        "airbnb",
+        "prettier" 
+    ],
+    "parser": "@typescript-eslint/parser",
+    "parserOptions": {
+        "ecmaFeatures": {
+            "jsx": true
+        },
+        "ecmaVersion": 12,
+        "sourceType": "module"
+    },
+    "plugins": [
+        "react",
+        "react-hooks",
+        "@typescript-eslint"
+    ],
+    "rules": {
+        "react/jsx-filename-extension": [ "warn", {"extensions": [".tsx"]} ],
+        "react-hooks/rules-of-hooks": "error",
+        "react-hooks/exhaustive-deps": "warn"
+    }
```

Comment:
I am also unsure about whether should we use these configurations or extend the inbuilt next or a mixture of both.
I personally feel that `eslint-config-next` should be extended as it contains some rules specific to next. For rest of the things, I am unsure.
Maybe can you look at some of the eslint configs for typescript and next and paste same links so we can see what's best

Reply:
I'll look at the eslint configs for typescript and next. Maybe we can do both.